tests/tools/test_openroad.py

```
@@ -4351,3 +4351,72 @@ def test_openroad_mode_sdcfileset_is_per_node(asic_gcd, tmp_path):
+    assert not any("asic,libcornerfileset,typical,nldm" in r for r in require), require
+
+
+def test_openroad_unsupported_delaymodel(asic_gcd):
+    # OpenROAD reads liberty through OpenSTA, so it can only be pointed at nldm or
+    # ccs filesets; anything else must fail setup instead of feeding it unreadable
+    # files.
+    _select_delaymodel(asic_gcd, "ecsm")
+
+    with pytest.raises(ValueError,
+                       match="^ecsm is not a supported delay model, "
+                             "supported delay models are: nldm, ccs$"):
```

Ruff (0.16.2)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)